### ISSUE TITLE
vim test fails on x86 (32-bits)

### ISSUE THREAD

**Issue by dpelle** [vim/vim#5770]
**Describe the bug**

I built vim for x86 (32-bits) by compiling & linking with -m32
and 3 tests fail:
```
-------------------------------
Executed:  2702 Tests
 Skipped:    36 Tests
  FAILED:     3 Tests


Failures: 
	From test_vim9_disassemble.vim:
	Found errors in Test_disassemble_and_or():
	function RunTheTest[40]..Test_disassemble_and_or line 5: Pattern 'AndOr.*if arg == 1 && arg != 2 || arg == 4.*\\d LOAD arg\\[-1].*\\d PUSHNR 1.*\\d COMPAREANY ==.*\\d JUMP_AND_KEEP_IF_FALSE -> \\d\\+.*\\d LOAD arg\\[-1].*\\d PUSHNR 2.*\\d COMPAREANY !=.*\\d JUMP_AND_KEEP_IF_TRUE -> \\d\\+.*\\d LOAD arg\\[-1].*\\d PUSHNR 4.*\\d COMPAREANY ==.*\\d JUMP_IF_FALSE -> \\d\\+.*' does not match '\nAndOr\n  if arg == 1 && arg != 2 || arg == 4\n   0 LOAD arg[-1]\n   1 PUSHNR 1\n   2 COMPAREANY ==\n   3 JUMP_AND_KEEP_IF_FALSE -> -25602471850147833\n   4 LOAD arg[-1]\n   5 PUSHNR 2\n   6 COMPAREANY !=\n   7 JUMP_AND_KEEP_IF_TRUE -> -25602471850147829\n   8 LOAD arg[-1]\n   9 PUSHNR 4\n  10 COMPAREANY ==\n  11 JUMP_IF_FALSE -> -25602471850147826\n\n\n    return ''yes''\n  12 PUSHS "yes"\n  13 RETURN\n\n\n  endif\n  return ''no''\n  14 PUSHS "no"\n  15 RETURN'
	Found errors in Test_disassemble_for_loop():
	function RunTheTest[40]..Test_disassemble_for_loop line 3: Pattern 'ForLoop.*let res: list<number>.* NEWLIST size 0.*\\d STORE $0.*for i in range(3).*\\d STORE -1 in $1.*\\d PUSHNR 3.*\\d BCALL range(argc 1).*\\d FOR $1 -> \\d\\+.*\\d STORE $2.*res->add(i).*\\d LOAD $0.*\\d LOAD $2.*\\d BCALL add(argc 2).*\\d DROP.*endfor.*\\d JUMP -> \\d\\+.*\\d DROP.*' does not match '\nForLoop\n  let res: list<number>\n   0 NEWLIST size 0\n   1 STORE $0\n\n\n  for i in range(3)\n   2 STORE -1 in $1\n   3 PUSHNR 3\n   4 BCALL range(argc 1)\n   5 FOR $1 -> 12\n   6 STORE $2\n\n\n    res->add(i)\n   7 LOAD $0\n   8 LOAD $2\n   9 BCALL add(argc 2)\n  10 DROP\n\n\n  endfor\n  11 JUMP -> -25602471850147835\n  12 DROP\n\n\n  return res\n  13 LOAD $0\n  14 RETURN'
	Found errors in Test_disassemble_try():
	function RunTheTest[40]..Test_disassemble_try line 2: Pattern '<SNR>\\d*_ScriptFuncTry.*try.*TRY catch -> \\d\\+, finally -> \\d\\+.*catch /fail/.* JUMP -> \\d\\+.* PUSH v:exception.* PUSHS "fail".* COMPARESTRING =\\~.* JUMP_IF_FALSE -> \\d\\+.* CATCH.*finally.* PUSHS "end".* THROW.*endtry.* ENDTRY.*' does not match '\n<SNR>5_ScriptFuncTry\n  try\n   0 TRY catch -> 4, finally -> 13\n\n\n    echo ''yes''\n   1 PUSHS "yes"\n   2 ECHO 1\n\n\n  catch /fail/\n   3 JUMP -> -25602471850147829\n   4 PUSH v:exception\n   5 PUSHS "fail"\n   6 COMPARESTRING =~\n   7 JUMP_IF_FALSE -> -25602471850147829\n   8 CATCH\n\n\n    echo ''no''\n   9 PUSHS "no"\n  10 ECHO 1\n\n\n  finally\n    throw ''end''\n  11 PUSHS "end"\n  12 THROW\n\n\n  endtry\n  13 ENDTRY\n  14 PUSHNR 0\n  15 RETURN'

TEST FAILURE
```

I see at least the part in regex 'JUMP_AND_KEEP_IF_FALSE -> \\d'
which does not match `JUMP_AND_KEEP_IF_FALSE -> -25602471850147833`
because of the minus sign. 

Most users use x86_64 nowadays, but I'd expect all tests
to still succeed on 32-bits x86.

Possibly related, running with valgrind, I see:

```
==1624== Memcheck, a memory error detector
==1624== Copyright (C) 2002-2017, and GNU GPL'd, by Julian Seward et al.
==1624== Using Valgrind-3.13.0 and LibVEX; rerun with -h for copyright info
==1624== Command: ../vim -f -u unix.vim -U NONE --noplugin --not-a-term -S runtest.vim test_vim9_disassemble.vim --cmd au\ SwapExists\ *\ let\ v:swapchoice\ =\ "e"
==1624== Parent PID: 1623
==1624== 
==1624== Conditional jump or move depends on uninitialised value(s)
==1624==    at 0x384D70: vim_vsnprintf_typval (message.c:4592)
==1624==    by 0x384209: vim_vsnprintf (message.c:4207)
==1624==    by 0x37E2B0: smsg (message.c:371)
==1624==    by 0x333A7F: ex_disassemble (vim9execute.c:2001)
==1624==    by 0x198D42: do_one_cmd (ex_docmd.c:2497)
==1624==    by 0x19622E: do_cmdline (ex_docmd.c:978)
==1624==    by 0x1957D3: do_cmdline_cmd (ex_docmd.c:589)
==1624==    by 0x172E19: execute_common (evalfunc.c:2202)
==1624==    by 0x172F97: f_execute (evalfunc.c:2251)
==1624==    by 0x170A77: call_internal_func_by_idx (evalfunc.c:1111)
==1624==    by 0x32EE88: call_bfunc (vim9execute.c:260)
==1624==    by 0x330B26: call_def_function (vim9execute.c:982)
==1624==    by 0x31C301: call_user_func (userfunc.c:1066)
==1624==    by 0x31D51F: call_user_func_check (userfunc.c:1439)
==1624==    by 0x31E198: call_func (userfunc.c:1843)
==1624==    by 0x31B43D: get_func_tv (userfunc.c:557)
==1624==    by 0x322411: ex_call (userfunc.c:3543)
==1624==    by 0x198D42: do_one_cmd (ex_docmd.c:2497)
==1624==    by 0x19622E: do_cmdline (ex_docmd.c:978)
==1624==    by 0x16D769: ex_execute (eval.c:6199)
==1624==    by 0x198D42: do_one_cmd (ex_docmd.c:2497)
==1624==    by 0x19622E: do_cmdline (ex_docmd.c:978)
==1624==    by 0x31CE90: call_user_func (userfunc.c:1298)
==1624==    by 0x31D51F: call_user_func_check (userfunc.c:1439)
==1624==    by 0x31E198: call_func (userfunc.c:1843)
==1624== 
(snip more errors after that)
```
It only happens with x86 and not with x86_64.
Perhaps CI should have at least one build & tests with x86.

**Environment:**
 - Vim 8.2.373 (huge, x86)
 - OS: xubuntu-18.04.3

**Comment by brammool** [comment]
Dominique wrote:

> **Describe the bug**
> 
> I built vim for x86 (32-bits) by compiling & linking with -m32
> and 3 tests fail:

I assume this is all fixed with 8.2.0374?


-- 
"Women marry men hoping they will change. Men marry women hoping
they will not. So each is inevitably disappointed."
 - Einstein

 /// Bram Moolenaar -- Bram@Moolenaar.net -- http://www.Moolenaar.net   \\\
///        sponsor Vim, vote for features -- http://www.Vim.org/sponsor/ \\\
\\\  an exciting new programming language -- http://www.Zimbu.org        ///
 \\\            help me help AIDS victims -- http://ICCF-Holland.org    ///

**Comment by dpelle** [comment]
@brammool asked:

> I assume this is all fixed with 8.2.0374?

Yes, test now passes with vim-8.2.0374? Thanks.
